src/ape/api/accounts.py

```diff
@@ -60,18 +60,9 @@ def call(self, txn: TransactionAPI, send_everything: bool = False) -> ReceiptAPI
         elif txn.nonce < self.nonce:
             raise AccountsError("Invalid nonce, will not publish.")
 
-        # TODO: Add `GasEstimationAPI`
-        if txn.gas_price is None:
-            txn.gas_price = self.provider.gas_price
-        # else: assume user specified a correct price, or will take too much time to confirm
-
-        # NOTE: Allow overriding gas limit
-        if txn.gas_limit is None:
-            txn.gas_limit = self.provider.estimate_gas_cost(txn)
-        # else: assume user specified the correct amount or txn will fail and waste gas
-
+        txn.set_defaults(self.provider)
```

Comment:
The above logic now happens in implementation's call of the `set_defaults(provider)` method.